### ISSUE TITLE
[Feature request] Allow campaign forms the option to send form contents to an address

### ISSUE THREAD

**Issue by RCheesley** [mautic/mautic#1121]
Sometimes with forms in a campaign, there might be content which is not saved to the database that needs emailing to a specified contact.

A standalone form has an action of sending the form contents to an address, but it does not currently seem possible to do that from a campaign form.

**Comment by alanhartless** [comment]
Yea, we're planning to bring back some of the form actions that make sense to the form campaign such as the send form results email, download asset, etc. Actually after some feedback on the confusion of form types (standalone vs campaign), we think we're going to simplify it.

Basically, we'll drop form types again and just have one type of form.  Then for form actions, ONLY have actions that make sense at the time of a form submit such as send form results email, asset download, send email to user, and tagging. All forms will then be available for use as a campaign lead source and any other action (add to lead list, etc) should be done via campaign.  What are your thoughts @RCheesley?

The tricky part will be migrating existing stand-alone form actions but we'll figure something out if we take this path.

**Comment by RCheesley** [comment]
I'm not sure it's a great idea to change back to the single form having
made people go single form -> two form types -> single form.  I think if
you just make it clearer what the forms are for, it will be fine.  Once I
understood the difference between the form types, all was fine, but that
process was in no way clear to me when I was trying to work it out myself.
There must have been a good reason for you to split into two form types,
what has happened to change that now?

Can't you just add the option to send form contents as a campaign workflow
form submission option?  My concern is that "actions that make sense at the
time of a form submit" could be pretty subjective, right?

*Ruth Cheesley*
*CEO & Co-Founder*

Virya Group Limited

P: +44 (0) 1473 852130
E: ruth.cheesley@viryagroup.com
M: +44 (0) 7920 404314

On 17 December 2015 at 18:50, Alan Hartless <notifications@github.com>
wrote:

> Yea, we're planning to bring back some of the form actions that make sense
> to the form campaign such as the send form results email, download asset,
> etc. Actually after some feedback on the confusion of form types
> (standalone vs campaign), we think we're going to simplify it.
>
> Basically, we'll drop form types again and just have one type of form.
> Then for form actions, ONLY have actions that make sense at the time of a
> form submit such as send form results email, asset download, send email to
> user, and tagging. All forms will then be available for use as a campaign
> lead source and any other action (add to lead list, etc) should be done via
> campaign. What are your thoughts @RCheesley <https://github.com/RCheesley>
> ?
>
> The tricky part will be migrating existing stand-alone form actions but
> we'll figure something out if we take this path.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mautic/mautic/issues/1121#issuecomment-165546012>.
>

-- 
Be like me, be Carbon free - don't print this and save a tree
IMPORTANT: The contents of this email and any attachments are confidential. 
They are intended for the named recipient(s) only. If you have received 
this email by mistake, please notify the sender immediately and do not 
disclose the contents to anyone or make copies thereof.

**Comment by alanhartless** [comment]
The reason for the switch to begin with was the form submit actions.  It was confusing people cause they were duplicating actions between form submit actions and campaign actions.  So we decided to differentiate into two types and only allow submit actions for the "standalone."  But then we found that some form submit actions still made sense for a campaign form such as sending a form results email or immediately downloading an asset (which can't be done through a campaign since it's executed outside the form submit request).  So that's why were thinking of dumping the differentiation again and just have form submit actions that can only be done at the time of a form submit and push everything else into a campaign.

Some things just can't be done in a campaign.  For example, offering an asset for download.  To do that through a campaign, you'd have to send an email with a link to the asset or have the asset attached.  If all is ok with that limitation, we could create a send form results campaign action. I know there's been a request out there to have a "send email to user" campaign action for a while too. 

Going to pull in @dbhurley to get his thoughts.
